mulog-cloudwatch/src/com/brunobonacci/mulog/publishers/cloudwatch.clj

```diff
@@ -0,0 +1,106 @@
+(ns com.brunobonacci.mulog.publishers.cloudwatch
+  (:require [com.brunobonacci.mulog.publisher :as p]
+            [com.brunobonacci.mulog.buffer :as rb]
+            [com.brunobonacci.mulog.utils :as ut]
+            [cognitect.aws.client.api :as aws]
+            [cheshire.core :as json]
+            [cheshire.generate :as gen]))
+
+
+(defn- has-failures?
+  [rs]
+  (not (= {} rs)))
+
+
+(defn- create-cloudwatch-client
+  [params]
+  (aws/client params))
+
+
+(defn- publish!
+  [cloudwatch-client group-name stream-name records]
+  (let [create-rs (aws/invoke cloudwatch-client {:op :CreateLogStream
+                                                 :request {:logGroupName group-name
+                                                           :logStreamName stream-name}})]
+    (if (has-failures? create-rs)
+      (throw
+        (ex-info
+          (str "μ/log cloudwatch publisher failure, group '" group-name "'" " stream '"  stream-name "'")
+          {:rs create-rs}))
+      (aws/invoke cloudwatch-client {:op      :PutLogEvents
+                                     :request {:logGroupName  group-name
+                                               :logStreamName stream-name
+                                               :logEvents     records}
+                                     }))))
+
+
+;;
+;; Add Flake encoder to JSON generator
+;;
+(gen/add-encoder com.brunobonacci.mulog.core.Flake
+                 (fn [x ^com.fasterxml.jackson.core.JsonGenerator json]
+                   (gen/write-string json ^String (str x))))
+
+
+
+(defn- put-log-events
+  [cw-client {:keys [group-name key-field format] :as config} records]
+  (let [fmt* (if (= :json format) json/generate-string ut/edn-str)
```

Comment:
does cloudwatch logs supports EDN as well? i don't think so.

Reply:
It works the same for `edn` but I guess `:pretty? true` should be enabled by default.
<img width="1287" alt="edn" src="https://user-images.githubusercontent.com/3762201/86940780-1a7f4100-c13b-11ea-9360-e4f48c554bc7.png">

Reply:
yes, it is just visualizing strings....
but can you query the data via Insights? if you have a numerical attribute can you aggregate them?

Reply:
I've checked and seems that `edn` is not really supported, also I tried to play with `parse` function which insight has but it doesn't look very useful comparing to `json`.
Should I simplify the code and leave `json` as the only option?

Reply:
I would say: yes! 
let's just use json